### ISSUE TITLE
Send Email asynchronously

### ISSUE THREAD

**Issue by jorge-leon** [django-helpdesk/django-helpdesk#1183]
**Is your feature request related to a problem? Please describe.**

Currently Emails are sent synchronously when posting or modifying a ticket.  This poses two problems:
- If Email sending is slow it is noticed directly by the end user.
- If Email sending fails, the end user gets a failure page.  The ticket is created anyway. The user has no
  clear indication how to recover correctly from the situation.

**Describe the solution you'd like**

Use a mail queuing module like [Django Mail Queue](https://pypi.org/project/django-mail-queue/), which
stores to be sent Emails in the database and allows for asynchronous delivery.

**Describe alternatives you've considered**

The case of failed Email sending could be handled gracefully with sensible user feedback.  However it burdens
the end user with a task that should be handled by the system administrator of the django-helpdesk installation.

**Additional context**

I might provide a PR at a later point of time.

**Comment by timthelion** [comment]
In my humble opinion this is out of scope of the project. We use `django-post-office` in production to this effect and that requires,no changes to helpdesk. You can just set your email backend like so

```
EMAIL_BACKEND = "post_office.EmailBackend"
    POST_OFFICE = {
        "BACKENDS": {
            "default": "django_ses.SESBackend",
        },
    }
```

For us it would be quite annoying to have Django Mail Queue baked into gelpdesk as then there would be a longer debug chain and more entries in the DB.

**Comment by jorge-leon** [comment]
You're right, that's more appropriate. Any queuing EMAIL_BACKEND will resolve the stated problems.

That would make this feature request only documentation relevant.

**Comment by uhurusurfa** [comment]
Closing as it seems there is agreement on how this issue should be handled.
